src/common/dependency-manager/local-manager.ts

```diff
@@ -53,24 +53,23 @@ export default class LocalDependencyManager extends DependencyManager {
     return PortUtil.getAvailablePort();
   }
 
-  async loadLocalService(service_path: string): Promise<DependencyNode> {
+  async loadLocalService(service_path: string): Promise<ServiceNode> {
     const config = ServiceConfigBuilder.buildFromPath(service_path);
 
     if (this.graph.nodes_map.has(`${config.getName()}:latest`)) {
       // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-      return this.graph.nodes_map.get(`${config.getName()}:latest`)!;
+      return this.graph.nodes_map.get(`${config.getName()}:latest`)! as ServiceNode;
```

Comment:
How will we know that this is a `ServiceNode`? Are we relying on the caller to know that `service_path` happens to refer to a ServiceNode and not some other node type?

Could we do a check for `__type` and throw if something is weird?

Reply:
For an environment a service:tag is unique so I'm relying on the caller.